### PR TITLE
Extension Manager

### PR DESCRIPTION
**Pull request by fath11**
feat:
- the ability to delete extensions

unfinished:
- reordering extensions, this PR won't be set as a draft because this function is commented out in the code. When it is ready for release another PR will be made. The reason for this is that I meant to implement reordering before opening a PR but due to high demands, I decided to open this PR first.

### CODE REVIEW

**Comment by zxq142857** [comment]
This idea is great. We recently discussed making a plugin like this.

File: src/plugins/plugins-manager/index.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ import styles from "./styles.less";
 const ALL_PLUGINS = Object.keys(Plugins);
 // When the line numbers of the variables below change, plopfile.js needs to be updated.
 const DEFAULT_INJECT_PLUGINS = [
+  "extension-manager",
```
Comment (zxq142857) [inline src/plugins/plugins-manager/index.tsx:14]: This plugin should not be installed by default.

File: src/plugins/extension-manager/styles.less [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+  font-size: 14px;
+  -webkit-user-select: none;
+  user-select: none;
+}
+
+.extensionManager-item_delete {
+  background-color: transparent; 
+  border-radius: 5px;
+  padding: 3px;
+  padding-bottom: 1px;
```
Comment (zxq142857) [inline src/plugins/extension-manager/styles.less:35]: "Setting it to 0 will make it look more centered vertically.

File: src/plugins/extension-manager/index.tsx [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+interface ExtensionManager {
+  _loadedExtensions: Map<string, string>;
+  deleteExtensionById: (id: string) => void;
+}
+
+interface VirtualMachineWithExtensions extends VirtualMachine {
+  extensionManager: ExtensionManager;
+}
+
+const ExtensionManager: React.FC<PluginContext & { vm: VirtualMachineWithExtensions }> = ({ intl, utils, vm, redux }) => {
```
Comment (zxq142857) [inline src/plugins/extension-manager/index.tsx:42]: I will complete the TS definitions for the VM as soon as possible.

File: src/plugins/extension-manager/index.tsx [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+  //console.log(vm)
+  //console.log(redux)
+
+  const locale = (redux.state.locales as any).locale
+
+  //patch the built-in getExtensionInfoByID
+  const getExtensionInfoById = (id: string) => {
```
Comment (zxq142857) [inline src/plugins/extension-manager/index.tsx:49]: This still doesn't perfectly solve the issue of translating the names of built-in plugins.

File: src/plugins/extension-manager/index.tsx [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    let info = (vm as any).extensionManager.getExtensionInfoById(id);
+
+    if (!info) {
+      info = {info: {extensionId: id, name: id}}
```
Comment (zxq142857) [inline src/plugins/extension-manager/index.tsx:53]: Replace `info:·{·extensionId:·id,·name:·id·}}` with `·info:·{·extensionId:·id,·name:·id·}·};`

**Review by zxq142857 — COMMENTED** [review]
It is still necessary to ensure proper code formatting checks.

**Comment by zxq142857** [comment]
After reviewing your code, it seems you've encountered some challenging issues. You can't retrieve information for all plugins (including the six built-in extensions and custom extensions). Therefore, I think an API should be provided to solve this problem.

**Comment by zxq142857** [comment]
The translation of the plugin name still needs to be added to l10n. In Chinese, it can be called "插件管理", with a description: "管理你所有已安装的插件" (Manage all your installed plugins).

File: src/plugins/extension-manager/index.tsx (hunk 4 above)
Comment (zxq142857) [reply src/plugins/extension-manager/index.tsx:49]: > This still doesn't perfectly solve the issue of translating the names of built-in plugins.

You can now get all extension information through the getAllExtensionInfo method in utils.

**Comment by fath11** [comment]
New update: all the dragging stuff are removed and multi select is a thing now

File: src/l10n/zh-cn.json [hunk 6]
```diff
@@ -188,5 +188,7 @@
   "plugins.customCss.title":"自定义CSS",
   "plugins.customCss.description":"加载自己的CSS文件！自定义你的Gandi样式！",
   "plugins.customCss.load":"从URL加载",
-  "plugins.customCss.load.description":"为了防止堆积，仅支持加载一个文件，如需加载多个，请合并"
+  "plugins.customCss.load.description":"为了防止堆积，仅支持加载一个文件，如需加载多个，请合并",
+  "plugins.ExtensionManager.title": "插件管理",
+  "plugins.ExtensionManager.intro": "管理你所有已安装的插件。"
```
Comment (MoreBugOfDog) [inline src/l10n/zh-cn.json:193]: image
This plugin is used to manage the installation and uninstallation of extensions
So I10n should read "拓展" not “插件”

**Review by MoreBugOfDog — CHANGES_REQUESTED** [review]

File: src/l10n/zh-cn.json (hunk 6 above)
Comment (fath11) [reply src/l10n/zh-cn.json:193]: zxq made the L10n, i just copy paste what he said above

**Pull request merged**